authors/apps/authentication/views.py

```diff
@@ -72,3 +81,61 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class ResetPasswordEmail(CreateAPIView):
+    """Password reset emailing with uid and reset token"""
+    permission_classes = (AllowAny,)
+    serializer_class = ResetEmailSerializer
+
+    @classmethod
+    def post(cls, request, *args, **kwargs):
+        """Returns message for reset password email"""
+        data = request.data
+        serializer = cls.serializer_class(data=data)
+        serializer.is_valid(raise_exception=True)
+        user = User.objects.get(email=data['email'])
+        body = json.dumps({
+            'user': data['email'],
+            'message': 'Please use the uid and token to rest your password',
+            'uid': urlsafe_base64_encode(force_bytes(data['email'])).decode(),
+            # 'token': default_token_generator.make_token(user),
+            'token': password_rest_token.make_token(user),
+        })
+        from_email = settings.DEFAULT_FROM_EMAIL
+        to_email = data['email']
+        subject = 'Confirm Your Article Account Password Reset'
+        try:
+            send_mail(subject, body, from_email, [to_email], fail_silently=False)
+            response = json.dumps({'message': 'check your email for the rest password token',
+                                   'status_code': status.HTTP_200_OK})
+        except Exception:
+            response = json.dumps({'Connection error': 'Check your internet connection',
+                                   'status_code': status.HTTP_503_SERVICE_UNAVAILABLE})
+        return Response(response, content_type='text/json')
+
+
+class ResetPasswordConfirm(RetrieveUpdateAPIView):
+    """Password resetting with a provided uid and token"""
+    permission_classes = (AllowAny,)
+    serializer_class = PasswordResetSerializer
+
+    @classmethod
+    def patch(cls, request, uidb64, token):
+        """Returns Password reset success"""
+        try:
+            uid = force_text(urlsafe_base64_decode(uidb64))
+            user = User.objects.get(email=uid)
+            if user is not None and password_rest_token.check_token(user, token):
+                data = request.data
+                serializer = cls.serializer_class(data=data)
+                serializer.is_valid(raise_exception=True)
+                user.password=data['password1']
+                user.save()
+                response = json.dumps({'message': 'Password successfully updated',
+                                         'status_code': status.HTTP_200_OK})
+        except(TypeError, ValueError, OverflowError, User.DoesNotExist):
+            user = None
+            response = json.dumps({'error': 'Password reset failed',
```

Comment:
@ManuelDominic , there's is no need to json load the response, Django rest Response will do that for you